src/autolabel/schema.py

```diff
@@ -71,6 +71,22 @@ class MetricResult(BaseModel):
     value: Any
 
 
+class ErrorType(str, Enum):
+    """Enum of supported error types"""
+
+    LLM_PROVIDER_ERROR = "llm_provider_error"
```

Comment:
do we have a way to understand what is the LLM provider error ? e.g. rate limit / availability errors can be fixed with retries but context length errors are deterministic. 

if this is hard to parse and needs provider specific logic, okay to club into one for now but ideally we can bubble the specific error to the users.

Reply:
I think we should definitely handle context length errors differently. They are handled separately for different providers and there is no consistent langchain error that is sent. We can do better and combine this across providers. Might need some custom string matching on the error for different providers - tracking this as a separate ticket https://github.com/refuel-ai/autolabel/issues/481